collection-scripts/gather

```diff
@@ -1,24 +1,66 @@
 #!/bin/bash
+# shellcheck disable=SC2154,SC1091,SC2086
+source common.sh
 
-# generate /must-gather/version file
-. version
-echo "opendatahub/must-gather" > /must-gather/version
-version >> /must-gather/version
+# for new flag to get since and since-time
+get_log_collection_args
 
+
+# generate /must-gather/version file with RHOAI version
+echo "red-hat-data-services/must-gather" > /must-gather/version
+rhoai_version >> /must-gather/version
+
+# define default namespaces
 OPERATOR_NS=${OPERATOR_NAMESPACE:-redhat-ods-operator}
 NOTEBOOKS_NS=${NOTEBOOK_NAMESPACE:-rhods-notebooks}
 MONITORING_NS=${MONITORING_NAMESPACE:-redhat-ods-monitoring}
 APPLICATIONS_NS=${APPLICATIONS_NAMESPACE:-redhat-ods-applications}
 
-oc adm inspect namespace/$OPERATOR_NS --dest-dir=must-gather || echo "Error getting logs from ${OPERATOR_NS}"
-oc adm inspect namespace/$NOTEBOOKS_NS --dest-dir=must-gather || echo "Error getting logs from ${NOTEBOOKS_NS}"
-oc adm inspect namespace/$MONITORING_NS --dest-dir=must-gather  || echo "Error getting logs from ${MONITORING_NS}"
-oc adm inspect namespace/$APPLICATIONS_NS --dest-dir=must-gather  || echo "Error getting logs from ${APPLICATIONS_NS}"
+oc adm inspect $log_collection_args "namespace/$OPERATOR_NS" --dest-dir="$DST_DIR" || echo "Error getting logs from ${OPERATOR_NS}"
+oc adm inspect $log_collection_args "namespace/$NOTEBOOKS_NS" --dest-dir="$DST_DIR" || echo "Error getting logs from ${NOTEBOOKS_NS}"
+oc adm inspect $log_collection_args "namespace/$MONITORING_NS" --dest-dir="$DST_DIR"  || echo "Error getting logs from ${MONITORING_NS}"
+oc adm inspect $log_collection_args "namespace/$APPLICATIONS_NS" --dest-dir="$DST_DIR"  || echo "Error getting logs from ${APPLICATIONS_NS}"
 
-/usr/bin/gather_data_science_pipelines
-/usr/bin/gather_model_mesh
-/usr/bin/gather_notebooks
-# force disk flush to ensure that all data gathered is accessible in the copy container
-sync
+# add DSCI and DSC
+get_operator_resource dscinitialization
+get_operator_resource datasciencecluster
 
+# set component or default to all
+component=${COMPONENT:-all}
+case "$component" in
+    "dsp")
+        /usr/bin/gather_data_science_pipelines
```

Comment:
Can it use the script directory for the binaries? like `BIN_DIR="$(realpath $0)"`?

So, that I can run it locally just in case.

Reply:
maybe we should have a follow up issue to make must-gather work locally than only with image?

Reply:
ok